File: CAndCppStudy/核心代码/速用库/基础代码/C/dup2/dup2.c
#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>
#include <string.h>
#include <sys/types.h>
int main(int argc,char*argv[])
{
	char buffer[20];
	int p[2];
	memset(buffer,0,20);
	pipe(p);
	printf("hello");
	dup2(p[1],STDOUT_FILENO);
	printf("world");
	close(p[1]);
	read(p[0],buffer,15);
	int fd=open(argv[1],O_WRONLY|O_CREAT|O_TRUNC,0300);
	write(fd,buffer,15);
	close(fd);
	return 0;
}

File: CAndCppStudy/核心代码/速用库/基础代码/C/dup2/dup2.py
import sys
import os

def main()->int:
    fd=open("./1.txt","w")
    save_fd=os.dup(1)
    os.dup2(fd.fileno(),1)
    fd.close()
    print("hello world")
    os.dup2(save_fd,1)
    return 0

'''
def main()->int:
    f=open('1.txt','a')
    os.dup2(f.fileno(), sys.stdout.fileno())
    f.close()
    print('runoob')
    print('google')
    sys.stdout.flush()
    return 0
'''

if __name__=="__main__":
    sys.exit(main())

